apps/webcert/src/utils/certificate/getClientValidationErrors.ts

```diff
@@ -123,12 +132,51 @@ const getErrorsFromConfig = (id: string, config: CertificateDataConfigType, valu
   }
 
   switch (config.type) {
+    case ConfigTypes.UE_DATE:
+      if (value.type === CertificateDataValueType.DATE) {
+        const hasDateBeforeMinLimit = isDateBeforeLimit(config.minDate, value.date)
+
+        return hasDateBeforeMinLimit
+          ? [
+              validationErrorFactory({
+                text: `Ange ett datum som är tidigast ${config.minDate ?? ''}.`,
+                type: 'INVALID_FORMAT',
+                showAlways: true,
+              }),
+            ]
+          : []
+      }
+      break
     case ConfigTypes.UE_CHECKBOX_DATE_RANGE_LIST:
       if (value.type === CertificateDataValueType.DATE_RANGE_LIST) {
         const hasAnyOverlap = value.list.some((val) => getPeriodHasOverlap(value.list, val.id))
-        if (hasAnyOverlap) {
-          return [validationErrorFactory(OVERLAP_ERROR)]
-        }
+        const overlapErrors = hasAnyOverlap ? [validationErrorFactory(OVERLAP_ERROR)] : []
+
+        const fromViolationAgainstMinLimit = value.list
+          .filter((val) => isDateBeforeLimit(config.min, val.from))
+          .map((val) =>
+            getValidationErrorWithCustomField(
+              id,
+              `Ange ett datum som är tidigast ${config.min ?? ''}.`,
+              'INVALID_FORMAT',
+              true,
+              `${val.id}.from`
+            )
+          )
+
+        const toViolationAgainstMinLimit = value.list
+          .filter((val) => isDateBeforeLimit(config.min, val.to))
+          .map((val) =>
+            getValidationErrorWithCustomField(
+              id,
+              `Ange ett datum som är tidigast ${config.min ?? ''}.`,
+              'INVALID_FORMAT',
+              true,
+              `${val.id}.to`
+            )
+          )
```

Comment:
This duplication here could perhaps be removed with a less generic function than `getValidationErrorWithCustomField`?
 
We could name the error type more specific, like "DATE_TO_EARLY" or something similar, since the date format is not really invalid here.

There is a `max` added to the config, should that also be validated?

Reply:
I've asked förvaltningen if they want the same for max

Reply:
Good comments I fixed everything but the max since I'm waiting for a response.

Reply:
I thought about the `getValidationErrorWithCustomField` a bit more and I think we could use the existing `getValidationErrorFactory` function instead.

We could add something like:
```typescript
const getEarlyDateError = (id: string, field: string, min?: string) => {
  return getValidationErrorFactory(
    id,
    field
  )({ type: 'INVALID_FORMAT', text: `Ange ett datum som är tidigast ${min ?? ''}.`, showAlways: true })
}
```

And use it like this:
```typescript
const fromViolationAgainstMinLimit = value.list
          .filter((val) => isDateBeforeLimit(config.min, val.from))
          .map((val) => getEarlyDateError(id, `${val.id}.from`, config.min))
```

And UE_DATE could be simplified a bit more:
```typescript
case ConfigTypes.UE_DATE:
      if (value.type === CertificateDataValueType.DATE) {
        return isDateBeforeLimit(config.minDate, value.date) ? [getEarlyDateError(id, id, config.minDate)] : []
      }
      break
```

Reply:
Tried this but the field doesn't get set. Will keep looking into it

Reply:
Done and added max validation since I didn't get a response